### DISCUSSION TITLE
How to properly update Calendr?

### DISCUSSION THREAD

**Discussion by Svec-Tomas** [pakerwreah/Calendr#221] [Ideas]
I've been wondering for a while:

Is there a way to automate the app update process?

I currently rely on two apps, [Latest](https://max.codes/latest/) and [MacUpdater](https://www.corecode.io/macupdater/), to keep most of my software updated.

However, Calendr seems to be an exception. It requires a fully manual update process where I have to download the latest version from GitHub and replace the existing installation.

Is there a way to streamline this process?

Perhaps I'm missing something?

I noticed Calendr might benefit from implementing [Sparkle](https://github.com/sparkle-project/Sparkle). Could that be a solution?

**Comment by pakerwreah** [comment]
I've used Sparkle before, it's nice, but unnecessarily overwhelming to set up.
I'm trying to find a solution that simply checks my Github releases, without having to setup rsa keys and host appcast files anywhere, but no luck yet.
I'm actually gonna do some research today.

**Reply by pakerwreah** [reply]
btw, you can install it via homebrew now, so it becomes as simple as:
```shell
brew update && brew upgrade calendr
```

**Reply by Svec-Tomas** [reply]
Thanks, Carlos. This is a great way to keep apps updated for people using Homebrew. Unfortunately, I haven't been able to figure it out yet because I find command-line tools a bit challenging. Maybe someday I'll give it another try.

In any case, I hope you can find a way to simplify app updates for those of us who don't use Homebrew.

**Reply by pakerwreah** [reply]
In the next version you will only have to click a `Install` button in the notification, or if you tap the notification body it will open the `About` window, where you can also install or check for updates again if necessary.

The only caveat is that it will show a `Save` popup for you to confirm where to install the app. You won't have to change anything, just click `Install`.

That's necessary to keep the `sandbox` enabled. It only grants access for the app to modify a single file, which is nice in terms of security and prevents dumb developer mistakes like deleting all your apps. 👀

`Sandbox` is also mandatory if I want to deploy it to the App Store in the future.

Although this is not 100% automatic, it's already a huge improvement compared to: open Github, download the zip file, extract the app, move it to the Applications folder and relaunch the app manually.

I have everything working already, but I will probably wait until we have some other new features to release.

**Reply by Svec-Tomas** [reply]
This sounds fantastic! ❤️

**Reply by pakerwreah** [reply]
There's a [v1.12.2](https://github.com/pakerwreah/Calendr/releases/tag/v1.12.2) `pre-release` available if you want to test it.

It will "update" to the previous version while it's not officially released, but that's nice because we can see how the update works :)

**Reply by Svec-Tomas** [reply]
I can confirm that it works as expected. Thanks a lot! 👍🙏
